src/vidpf.rs

```diff
@@ -226,6 +228,67 @@ impl<W: VidpfValue, const NONCE_SIZE: usize> Vidpf<W, NONCE_SIZE> {
         })
     }
 
+    /// [`Vidpf::eval_cached`] evaluates the entire `input` and produces a share of the
+    /// input's weight. It reuses computation from previous levels available in the
+    /// cache
+    pub fn eval_cached(
```

Comment:
More rusty (here and below)
```suggestion
    pub fn eval_with_cache(
```